src/katgpucbf/xbgpu/main.py

```diff
@@ -48,13 +49,71 @@
 from ..spead import DEFAULT_PORT
 from ..utils import add_signal_handlers, parse_source
 from .correlation import device_filter
+from .output import BOutput, XOutput
 
 logger = logging.getLogger(__name__)
 
 
+class BOutputDict(TypedDict, total=False):
+    """Configuration options for a beam output.
+
+    Unlike :class:`BOutput`, all the fields are optional, so that it can
+    be built up incrementally. They must all be filled in before using it to
+    construct a :class:`BOutput`.
+    """
+
+    name: str
+    dst: Endpoint
+    channels_per_substream: int
```

Comment:
I know @bmerry does the F-engine OutputDicts in `main.py`, but I'm wondering whether they shouldn't be in `output.py` instead, just to keep this file a bit less cluttered.

Reply:
The reason for keeping it in main.py is that it's internal to the implementation of argument parsing, rather than a public API. In fact I think it may be worth using an underscore prefix to emphasise that it's not part of the public API of main.py (in which case, do the same in fgpu.main).

Reply:
Ok, that makes sense.

In which case I'd suggest making that a separate, technical-debt ticket. @amishatishpatel, ignore this comment for the purposes of this PR.

Reply:
[As requested](https://skaafrica.atlassian.net/browse/NGC-988).